validate_key.py

```diff
@@ -0,0 +1,56 @@
+from datetime import datetime, timedelta
+import hashlib
+
+from pywell.entry_points import run_from_cli, run_from_api_gateway
+
+
+DESCRIPTION = 'Validate a download key.'
+
+ARG_DEFINITIONS = {
+    'KEY': 'Key to validate.',
+    'MAX_AGE': 'Number of days a key should be considered valid.',
+    'SECRET': 'Secret to use for validation.'
+}
+
+REQUIRED_ARGS = ['KEY', 'MAX_AGE', 'SECRET']
+
+
+def main(args):
```

Comment:
can you call this `validate_key` -- it's confusing to call it main -- the function validates the key, and this isn't a module with other functions where we need to distinguish the library code from many other functions.

Reply:
Let's resolve this in the other PR, where it's already discussed and where this code is first added.